java-processors/src/main/java/com/dataiku/dip/plugins/tradearea/AreaGenerator.java

```diff
@@ -0,0 +1,146 @@
+package com.dataiku.dip.plugins.tradearea;
+
+import com.dataiku.dip.shaker.types.GeoPoint;
+import com.dataiku.dip.utils.DKULogger;
+
+/**
+ * Abstract class expected by TradeAreaProcessor to generate the polygons
+ */
+abstract class AreaGenerator {
+
+    private static DKULogger logger = DKULogger.getLogger("dku");
+
+    public abstract String generateArea(GeoPoint.Coords coords);
+
+    public static void checkInputParams(TradeAreaProcessor.UnitMode unitMode, TradeAreaProcessor.ShapeMode shapeMode,
+                                           double radius, double height, double width){
+
+        switch (shapeMode) {
+        case RECTANGLE:
+            if (width <= 0 || height <= 0) {
+                logger.info("Rectangle Area Generator: Received invalid parameters as input.");
+                logger.infoV("Got width= {}", width);
+                logger.infoV("Got height= {}", height);
+                throw new IllegalArgumentException(
+                        "Rectangular Trade Area Processor Failed.\n"+
+                        "The height and width of the trade area must be strictly greater than 0.\n"+
+                        "Received height value is equal to "+ height + "\n"+
+                        "and width value is equal to "+ width);
+            }
+            break;
+        case CIRCLE:
+            if (radius <= 0) {
+                logger.info("Circle Area Generator: Received invalid parameters as input.");
+                logger.infoV("Got radius= {}", radius);
+                throw new IllegalArgumentException(
+                        "Circular Trade Area Processor Failed.\n"+
+                        "The radius of the trade area must be strictly greater than 0.\n"+
+                        "Received radius value is equal to "+ radius);
+            }
+            break;
+        default:
+            throw new IllegalArgumentException("Invalid processing mode: " + shapeMode);
+        }
+    };
+
+}
+
+/**
+ * Generates a rectangular area centered on an input geopoint coordinates.
+ * The parameters width, height, radius are expected to be in kilometers.
+ * If using miles in the processor, those distances must be converted to miles before.
+ */
+class RectangleAreaGenerator extends AreaGenerator {
+
+    double width;
+    double height;
+    double radius;
+    double diagonalAngle;
+
+    private static DKULogger logger = DKULogger.getLogger("dku");
+
+    public RectangleAreaGenerator(double width, double height){
+        if (width <= 0 || height <= 0){
+            logger.info("Rectangle Area Generator: Received invalid parameters as input.");
+            logger.infoV("Got width= {}", width);
+            logger.infoV("Got height= {}", height);
```

Comment:
init this.height, this.width and this.diagonalAngle to 0

Reply:
Done in a8002c2